File: backend/src/main/java/me/storyfor/backend/service/ChildService.java
package me.storyfor.backend.service;

import me.storyfor.backend.dto.ChildDto;
import me.storyfor.backend.dto.ChildRequest;
import me.storyfor.backend.entity.Child;
import me.storyfor.backend.repository.ChildRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public class ChildService {

    private final ChildRepository childRepository;

    public ChildService(ChildRepository childRepository) {
        this.childRepository = childRepository;
    }

    public List<ChildDto> getChildren(UUID userId) {
        return childRepository.findByUserIdOrderByCreatedAtDesc(userId)
                .stream().map(ChildDto::from).toList();
    }

    public ChildDto createChild(UUID userId, ChildRequest request) {
        if (childRepository.countByUserId(userId) >= 5) {
            throw new IllegalStateException("Maximum 5 children allowed");
        }
        Child child = new Child();
        child.setUserId(userId);
        child.setName(request.name());
        child.setAge(request.age());
        child.setGender(request.gender());
        child.setInterests(request.interests());
        child.setFavoriteAnimal(request.favoriteAnimal());
        return ChildDto.from(childRepository.save(child));
    }

    public ChildDto updateChild(UUID userId, UUID childId, ChildRequest request) {
        Child child = childRepository.findById(childId)
                .orElseThrow(() -> new RuntimeException("Child not found"));
        if (!child.getUserId().equals(userId)) {
            throw new RuntimeException("Not authorized");
        }
        child.setName(request.name());
        child.setAge(request.age());
        child.setGender(request.gender());
        child.setInterests(request.interests());
        child.setFavoriteAnimal(request.favoriteAnimal());
        return ChildDto.from(childRepository.save(child));
    }

    public void deleteChild(UUID userId, UUID childId) {
        Child child = childRepository.findById(childId)
                .orElseThrow(() -> new RuntimeException("Child not found"));
        if (!child.getUserId().equals(userId)) {
            throw new RuntimeException("Not authorized");
        }
        childRepository.delete(child);
    }
}

File: frontend/src/app/services/child.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';

export interface Child {
  id: string;
  name: string;
  age: number;
  gender: string;
  interests: string[];
  favoriteAnimal: string;
}

@Injectable({ providedIn: 'root' })
export class ChildService {
  constructor(private api: ApiService) {}

  getChildren() {
    return this.api.get<Child[]>('/children');
  }

  createChild(child: Partial<Child>) {
    return this.api.post<Child>('/children', child);
  }

  updateChild(id: string, child: Partial<Child>) {
    return this.api.put<Child>(`/children/${id}`, child);
  }

  deleteChild(id: string) {
    return this.api.delete(`/children/${id}`);
  }
}
